packages/next/next-server/lib/router/utils/route-regex.ts

```diff
@@ -5,7 +5,8 @@ function escapeRegex(str: string) {
 }
 
 export function getRouteRegex(
-  normalizedRoute: string
+  normalizedRoute: string,
+  trailingSlash = true
```

Comment:
This code is bundled for the client side, so I'd prefer to avoid adding this logic to this method.

Reply:
Updated it to not have a default arg value and simplified the check so it should be adding a minimal amount of code to the client